src/JuliusSweetland.OptiKey.Core/UI/ViewModels/MainViewModel.ServiceEventHandlers.cs

```diff
@@ -2372,47 +2380,117 @@ private void SelectVoice(string voice)
             NavigateToMenu();
         }
 
-        private void StepList(string command)
+        public void StepList(KeyValue singleKeyValue, List<string> multiKeySelection)
+        {
+            if (!string.IsNullOrEmpty(singleKeyValue.String))
+            {
+                Log.InfoFormat("StepList called with singleKeyValue [{0}]", singleKeyValue.String);
+
+                List<string> commandList = singleKeyValue.String.Split('<').ToList();
+                if (commandList.Any())
+                {
+                    var cTask = Task.Run(() => StepListCommands(singleKeyValue, multiKeySelection, commandList));
```

Comment:
@JuliusSweetland I got it working so that I can start the threaded task at this point.

Reply:
@AdamRoden If you do this you are spinning up a new threadpool thread to run the StepListCommands method on. You will then have to dispatch back to the UI thread to perform any action on a key. I would think it better to keep everything on the UI thread, i.e. make the whole method chain async and then correctly await each call down the chain.